### PR TITLE
[MessageControl] Making the plugin work in-process only, separating Logging and Tracing functionality 

### PR DESCRIPTION
**Pull request by VeithMetro**
Needs both Thunder (https://github.com/rdkcentral/Thunder/pull/1312) and ThunderInterfaces (https://github.com/rdkcentral/ThunderInterfaces/pull/251) changes to be merged before this can be successfully built.

### CODE REVIEW

**Review by sebaszm — CHANGES_REQUESTED** [review]
See comments.

File: MessageControl/MessageControl.h [hunk 1]
```diff
@@ -248,30 +306,65 @@ namespace Plugin {
...
             _adminLock.Lock();
-            _collect->Detach(id);
+
+            _client.RemoveInstance(id);
+            _cleaning.emplace_back(id);
+            Cleanup();
+            
             _adminLock.Unlock();
 
             if (id == _connectionId) {
```
Comment (VeithMetro) [inline MessageControl/MessageControl.h:370]: Can we get rid of the `_connectionId` class member entirely then, if it is not need for `IRemoteConnection`, or does it need to stay for this part?

**Review by VeithMetro — COMMENTED** [review]
See comments

**Review by sebaszm — APPROVED** [review]

**Review by pwielders — APPROVED** [review]
I put my trust in @sebaszm :-)

**Pull request merged**